src/runtime/components/nuxt-icon.vue

```diff
@@ -1,8 +1,9 @@
 <template>
-  <span class="nuxt-icon" :class="{ 'nuxt-icon--fill': !filled }" v-html="rawIcon" />
+  <span class="nuxt-icon" :class="{ 'nuxt-icon--fill': !filled }" v-html="icon" />
```

Comment:
The [v-html](https://vuejs.org/api/built-in-directives.html#v-html) directive expects a `string`. What will happen if you will pass a `Record<string, any>` as you mentioned in a type declaration?